crates/bitwarden-c/src/c.rs

```diff
@@ -28,6 +33,48 @@ pub extern "C" fn run_command(c_str_ptr: *const c_char, client_ptr: *const CClie
     }
 }
 
+type OnCompletedCallback = unsafe extern "C" fn(result: *mut c_char) -> ();
+
+#[no_mangle]
+pub extern "C" fn run_command_async(
+    c_str_ptr: *const c_char,
+    client_ptr: *const CClient,
+    on_completed_callback: OnCompletedCallback,
+    is_cancellable: bool,
+) -> *mut JoinHandle<()> {
+    let client = unsafe { ffi_ref!(client_ptr) };
+    let input_str = str::from_utf8(unsafe { CStr::from_ptr(c_str_ptr) }.to_bytes())
+        .expect("Input should be a valid string")
+        // Languages may assume that the string is collectable as soon as this method exits
+        // but it's not since the request will be run in the background
+        // so we need to make our own copy.
+        .to_owned();
+
+    let join_handle = client.runtime.spawn(async move {
+        let result = client.client.run_command(input_str.as_str()).await;
+        let str_result = match std::ffi::CString::new(result) {
+            Ok(cstr) => cstr.into_raw(),
+            Err(_) => panic!("failed to return comment result: null encountered"),
+        };
+
+        // run completed function
+        unsafe {
+            on_completed_callback(str_result);
+            let _ = CString::from_raw(str_result);
+        }
+    });
+
+    // We only want to box the join handle the caller
+    // has said that they may want to cancel, essentially
+    // promising to us that they will take care of the
+    // returned pointer.
```

Comment:
This seems fairly compact, `cargo fmt` generally uses 100 as the rule for line breaks.

```suggestion
    // We only want to box the join handle the caller has said that they may want to cancel,
    // essentially promising to us that they will take care of the returned pointer.
```